packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.swift

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import Foundation
+
+public final class IosPlatformImagesPlugin: NSObject, FlutterPlugin {
+
+  private let channel: MethodChannel
+
+  init(channel: MethodChannel) {
+    self.channel = channel
+  }
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/ios_platform_images",
+      binaryMessenger: registrar.messenger())
+
+    let instance = IosPlatformImagesPlugin(channel: channel)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    switch call.method {
+    case "loadImage":
+      loadImage(call, result)
+    case "resolveURL":
+      resolveURL(call, result)
+    default:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+  private func loadImage(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
+    guard let arguments = call.arguments as? [Any],
+      let name = arguments.first as? String,
+      let image = UIImage.flutterImage(withName: name)
+    else {
+      result(nil)
+      return
+    }
+
+    let data = image.pngData()
+
+    let imageResult: [String: Any] = [
+      "scale": image.scale,
+      "data": FlutterStandardTypedData(bytes: data!),
+    ]
+
+    result(imageResult)
+  }
+
+  private func resolveURL(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
+    if let args = call.arguments as? [String?] {
+      let name = args[0]
+      let extensionName = args.count == 2 ? args[1] : nil
+
+      if let url = Bundle.main.url(forResource: name, withExtension: extensionName) {
```

Comment:
Why the explicit branching, rather than optional chaining like the Obj-C version? The more `result` calls there are, the harder it is to make sure the logic is correctly calling it exactly one time.